File: app/src/main/java/com/diarmaidlindsay/koohii/utils/ToastUtil.java
package com.diarmaidlindsay.koohii.utils;

import android.app.Activity;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import android.widget.Toast;
import com.diarmaidlindsay.koohii.R;

/**
 * Replacement for Android Toast class so I can programmatically apply themes
 */
public class ToastUtil {
    public static Toast makeText(Context context, String text, int duration) {
        LayoutInflater inflater = LayoutInflater.from(context);
        Activity activity = (Activity) context;
        View layout = inflater.inflate(R.layout.toast,
                (ViewGroup) activity.findViewById(R.id.custom_toast_layout));

        // set a message
        TextView textView = (TextView) layout.findViewById(R.id.toast_text);
        textView.setText(text);

        // Toast...
        Toast toast = new Toast(context);
        toast.setDuration(duration);
        toast.setView(layout);
        return toast;
    }

    public static Toast makeText(Context context, int resId, int duration) {
        return makeText(context, context.getResources().getText(resId).toString(), duration);
    }
}

File: app/src/main/java/tech/diarmaid/koohiiaite/utils/ToastUtil.kt
package tech.diarmaid.koohiiaite.utils

/**
 * Replacement for Android Toast class so I can programmatically apply themes
 */
object ToastUtil {
//    fun makeText(context: Context, text: String, duration: Int): Toast {
//        val inflater = LayoutInflater.from(context)
//        val activity = context as Activity
//        //TODO : This is broken, null pointer exception
//        val layout = inflater.inflate(R.layout.toast,
//                activity.findViewById<View>(R.id.custom_toast_layout) as ViewGroup)
//
//        // set a message
//        val textView = layout.findViewById<TextView>(R.id.toast_text)
//        textView.text = text
//
//        // Toast...
//        val toast = Toast(context)
//        toast.duration = duration
//        toast.view = layout
//        return toast
//    }
//
//    fun makeText(context: Context, resId: Int, duration: Int): Toast {
//        return makeText(context, context.resources.getText(resId).toString(), duration)
//    }
}
